Plugin/Block/Toolbar.php

```diff
@@ -0,0 +1,184 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2017 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Plugin\Block;
+
+use Nosto\Tagging\Plugin\Catalog\Model\Config;
+use Nosto\Tagging\Helper\Data as NostoHelperData;
+use Nosto\Tagging\Helper\Account as NostoHelperAccount;
+use Magento\Backend\Block\Template\Context;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Catalog\Block\Product\ProductList\Toolbar as MagentoToolbar;
+use Magento\Store\Model\Store;
+use Magento\Framework\Registry;
+use Nosto\Tagging\Model\Category\Builder as CategoryBuilder;
+use Magento\Framework\Stdlib\CookieManagerInterface;
+use Nosto\Tagging\Model\Customer\Customer as NostoCustomer;
+use Nosto\Tagging\Model\Service\Recommendation\Category as CategoryRecommendation;
+use Magento\Framework\Model\ResourceModel\Db\Collection\AbstractCollection;
+use Magento\Framework\Data\Collection;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Element\Template;
+use Nosto\Tagging\Logger\Logger as NostoLogger;
+
+class Toolbar extends Template
+{
+    const SORT_ORDER_DESC = 'DESC';
+
+    /**  @var StoreManagerInterface */
+    private $storeManager;
+
+    /** @var NostoHelperData */
+    private $nostoHelperData;
+
+    /** @var NostoHelperAccount */
+    private $nostoHelperAccount;
+
+    /**  @var CategoryBuilder */
+    private $categoryBuilder;
+
+    /** @var Registry */
+    private $registry;
+
+    /** @var CookieManagerInterface */
+    private $cookieManager;
+
+    /** @var CategoryRecommendation */
+    private $categoryRecommendaiton;
+
+    /** @var NostoLogger */
+    private $logger;
+
+    /** @var AbstractCollection */
+    protected $_collection = null;
```

Comment:
Why was this local variable introduced here if it's only used inside the `afterSetCollection` method?